agent/checks/alias.go

```diff
@@ -85,16 +85,22 @@ func (c *CheckAlias) runLocal(stopCh chan struct{}) {
 	c.Notify.AddAliasCheck(c.CheckID, c.ServiceID, notifyCh)
 	defer c.Notify.RemoveAliasCheck(c.CheckID, c.ServiceID)
 
+	updateStatus := func() {
+		checks := c.Notify.Checks()
+		checksList := make([]*structs.HealthCheck, 0, len(checks))
+		for _, chk := range checks {
+			checksList = append(checksList, chk)
+		}
+		c.processChecks(checksList)
+	}
+
+	// Immediately run to get the current state of the target service
+	updateStatus()
```

Comment:
This fixed an existing bug though - Alias checks before this never updated if their target had no checks (or was already registered and the checks never change).

I don't think it was a _known_ buf but might be worth a CHANGELOG entry anyway.